spritecontrol.cpp

```diff
@@ -2,19 +2,23 @@
 #include "resources.h"
 #include "config.h"
 #include "noise.h"
+#include "configdialog.h"
 
 using std::get;
 
 SpriteGenerator::SpriteGenerator() {
 	using namespace std::chrono;
 	std::srand(duration_cast<milliseconds>(system_clock::now().time_since_epoch()).count());
 
-	std::vector<int> bag_of_palettes(_PALETTE_COUNT);
-	std::iota(bag_of_palettes.begin(), bag_of_palettes.end(), 0);
+	//std::vector<int> bag_of_palettes(static_cast<int>(PaletteName::_PALETTE_COUNT));
+	//std::iota(bag_of_palettes.begin(), bag_of_palettes.end(), 0);
+	std::vector<PaletteName> bag_of_palettes = PALETTES_BY_GROUP.at((PaletteGroup) (cfg.at(YonkPalette)));
 
-	for (int i = 0; i < round(cfg.at(MaxColors)); i++) {
+	float maxColors = cfg.at(MaxColors) * (bag_of_palettes.size() / config_ranges.at(MaxColors).second);
+
+	for (int i = 0; i < round(maxColors); i++) {
```

Comment:
I'm scared of this. If the result of `round(maxColors)` is greater than `bag_of_palettes.size()`, the program will try to erase from an empty list and invoke undefined behavior. We cannot mathematically reason that this will never be the case because of floating-point imprecision. 

Since it's so important for this to not exceed `bag_of_palettes.size()`, I think we should do something like this, even though it is pretty clunky:

```c++
if (round(maxColors) > bag_of_palettes.size()) {
    maxColors = bag_of_palettes.size();
}
```